src/lib.rs

```diff
@@ -42,26 +42,48 @@ pub fn clockboard(
     //boundary: Option<Polygon<f64>>,
 ) -> GeoJson {
     let mut polygons = Vec::new();
-    for i in params.distances {
-        // println!("{}", i); // debugging
-        let circle = makecircle(centerpoint, i, params.num_vertices);
-        polygons.push(circle);
+    let mut irad_inner: f64 = 0.0;
+    if params.num_segments == 1 {
+        for i in params.distances {
+            let circle = makecircle(centerpoint, i, params.num_vertices);
+            polygons.push(circle);
+        }
+    } else {
+        for i in 0..params.distances.len() {
+            let irad = params.distances[i];
+            if i == 0 {
+                let irad_inner = 0.0;
+            } else {
+                let irad_inner = params.distances[(i - 1)];
+            }
+            for j in 0..params.num_segments {
+                let circle = clockpoly(
```

Comment:
This isn't a circle; it returns one of the zones. We could call it `zone` or even just `polygon`, unless you have another term for it.

Reply:
Agreed, sorted now.